### ISSUE TITLE
Ability to add videos from other sites

### ISSUE THREAD

**Issue by mgogoulos** [plumi/plumi.app#1006]
**Reported by and on 5 Jul 2007 16:16 UTC**
Plone4artists video allows you to take videos from Google Video, YouTube, blip.tv, etc and add them as an item.

http://plone4artists.org/products/plone4artistsvideo

It would be good to create a specific content type/product for adding this kind of videos to enable broader sharing of content.

**Comment by mgogoulos** [comment]
**Comment by anonymous on 19 Oct 2007 06:35 UTC**
Talking to someone from foe today they were looking at updating to the latest version of kupu which allows embedded videos. This is a bit simpler than what I was envisioning but could be good to have available.

Other possibilities include 
http://plone.org/products/atgooglevideo/

**Comment by mgogoulos** [comment]
**Modified by and on 29 Nov 2009 08:46 UTC**

**Comment by mgogoulos** [comment]
**Modified by and on 21 Apr 2010 03:50 UTC**

**Comment by mgogoulos** [comment]
**Modified by and on 21 Apr 2010 05:48 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 26 Oct 2010 13:15 UTC**

**Comment by mgogoulos** [comment]
**Modified by and on 26 Oct 2010 13:56 UTC**

**Comment by mgogoulos** [comment]
**Modified by and on 28 Oct 2010 10:54 UTC**

**Comment by mgogoulos** [comment]
**Comment by mike on 10 Nov 2010 10:07 UTC**
we need to think about how videos hosted on external sites will be displayed for the user.

the viewer of these videos will have a different experience to viewing a normal video - you will not be able to download it. we need to then explain in the UI that this is the case...

or should videos hosted externally be quite separate from internal videos?

- anna

**Comment by mgogoulos** [comment]
**Comment by anna on 2 Dec 2010 11:37 UTC**
Ok, we discussed how these videos might work. The proposal has been added to the ticket description.

**Comment by mgogoulos** [comment]
**Modified by anna on 2 Dec 2010 11:43 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 3 Dec 2010 01:21 UTC**
Mike suggests using the same PlumiVideo content type, with option to add link / embed code instead of upload a file (but still be able to differentiate these videos from normal videos, and thus list them differently on the site), and also to split off the changes to publish form into another ticket.

Ticket for changes to publish form:
http://plumi.org/ticket/529

**Comment by mgogoulos** [comment]
**Comment by anna on 15 Dec 2010 11:46 UTC**
Moving this to the re-factoring milestone, as Unweb advise that this should be done after changing Plumi video content type into schema extender

**Comment by mgogoulos** [comment]
**Modified by anna on 28 Jan 2011 04:40 UTC**

**Comment by mgogoulos** [comment]
**Comment by dimo on 13 Apr 2011 16:26 UTC**
This is not something that can be completed in this milestone & contract. My suggestion is to do it in parallel with the video publish form redesign

**Comment by mgogoulos** [comment]
**Comment by and on 15 Apr 2011 15:09 UTC**
I will convert the remaining 4.2 tickets to a 4.3 milestone and shift the rest back a version also. Major means it won't actually go in 4.2.

**Comment by mgogoulos** [comment]
**Modified by dimo on 28 Sep 2011 17:54 UTC**

**Comment by mgogoulos** [comment]
**Comment by and on 28 Sep 2011 18:59 UTC**
In previous conversations we discussed doing this as part of the publish form refactor so that it was already integrated, if we're doing the refactor in the 4.3 release shouldn't add it into 4.3?

**Comment by mgogoulos** [comment]
**Comment by anna on 24 Oct 2012 03:34 UTC**
This apparently can't happen in 4.5, but moving to 4.6 to indicate this as a priority.

**Comment by mgogoulos** [comment]
**Modified by and on 19 Mar 2013 23:09 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 22 Mar 2013 03:48 UTC**

**Comment by mgogoulos** [comment]
**Comment by anonymous on 23 Apr 2013 10:19 UTC**
I will use Plumi if this is implemented!

**Comment by mgogoulos** [comment]
**Comment by and on 23 Apr 2013 13:38 UTC**
Let us know if you can help!

**Comment by mgogoulos** [comment]
**Modified by anna on 25 Sep 2014 11:43 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 25 Sep 2014 11:44 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 3 Oct 2014 09:32 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 31 Oct 2014 11:18 UTC**
Testing feedback from Andrew:


* Is it possible to make the option to add the external video be part of adding an existing video? Ie. that the option is present to you at the top of the page?
* It seems that the meta data is the same as the old publish form rather than the new one. Can we not use the new system? It's quite difficult to add categories etc, license etc.
* Text at the top "The link to the video on the external site (on Vimeo, etc.)." I think should include YT also as an example
* Can you tell me about the listing of the videos in the categories? Will they appear in the country/topic etc with thumbnail and description?
* I realise this may be beyond what is possible now but have you investigated what it would take for amara to work on these videos?
* Do the videos appear on the author page? (I can't seem to access my admin privileged account on demo so can't test this - if you want to make test000 an admin that'd be great.
* Will the videos appear in the 'other videos by this member' section also?
* I couldn't find any licensing option in the form, eg. for CC

**Comment by mgogoulos** [comment]
**Comment by anna on 31 Oct 2014 11:19 UTC**
We would have to leave anything regarding Amara out of the question for now, but we should try to address all other points as listed above.

**Comment by mgogoulos** [comment]
**Modified by anna on 12 Dec 2014 03:34 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 16 Dec 2014 01:16 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 16 Dec 2014 07:42 UTC**
Cool, Sam has got the publish form working with toggle between video link and upload:
http://staging.engagemedia.org:9080/Members/anna/videos/@@publish_video

There are a number of issues to address of course, I'll log them in the description above.

**Comment by mgogoulos** [comment]
**Modified by anna on 16 Dec 2014 07:56 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 16 Dec 2014 08:00 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 16 Dec 2014 09:28 UTC**
Sam - can we try using some nice big buttons like this?

http://codepen.io/cheeriottis/pen/inluv

See customised version below..

```
<html lang="en">
<head>
  <meta name="author" content="www.twitter.com/cheeriottis">
  <link href='http://fonts.googleapis.com/css?family=Oswald' rel='stylesheet' type='text/css'>
  <link rel="stylesheet" href="css/core.css">
</head>
<body>
  <a href="#" class="action-button shadow animate blue">Upload File</a>
  <a href="#" class="action-button shadow animate red">Paste Link</a>
</body>
</html>
```



```
body
{
	padding: 50px;
}

.animate
{
	transition: all 0.1s;
	-webkit-transition: all 0.1s;
}

.action-button
{
	position: relative;
	padding: 10px 40px;
  margin: 0px 10px 10px 0px;
  float: left;
	border-radius: 10px;
	font-family: 'Oswald';
	font-size: 25px;
	color: #FFF;
	text-decoration: none;	
}

.blue
{
	background-color: #41B5E2;
	border-bottom: 5px solid #0E728B;
	text-shadow: 0px -2px #2980B9;
}

.red
{
	background-color: #0E728B;
	border-bottom: 5px solid #252830;
	text-shadow: 0px -2px #252830;
}

.green
{
	background-color: #82BF56;
	border-bottom: 5px solid #669644;
	text-shadow: 0px -2px #669644;
}

.yellow
{
	background-color: #F2CF66;
	border-bottom: 5px solid #D1B358;
	text-shadow: 0px -2px #D1B358;
}

.action-button:active
{
	transform: translate(0px,5px);
  -webkit-transform: translate(0px,5px);
	border-bottom: 1px solid;
}
```

**Comment by mgogoulos** [comment]
**Comment by anna on 16 Dec 2014 09:30 UTC**
See screenshot attached of nice big CSS3 buttons.

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 01:40 UTC**
Updated remaining tasks in description.

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 01:40 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 02:11 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 02:11 UTC**
buttons HTML:

  <p style="text-align: center;">
                <a href="#" data-mode="upload"   class="button addModeSelector" style="width: auto; display: inline-block; margin: 20px; padding: 20px; font-size: 160%;">Upload File</a>
                <a href="#" data-mode="external" class="button addModeSelector" style="width: auto; display: inline-block; margin: 20px; padding: 20px; font-size: 160%;">Paste Link</a>
                <input type="submit" style="visibility: hidden;"/>
1:12
              </p>

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 02:18 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 02:26 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 02:50 UTC**
found a UI bug. having initially chosen Upload File, and then filling out info on first tab, but then when clicking Save Changes on the Categorise page, the page reloads to show those two buttons again


it *has* remembered the categorisation fields, which is great, it just shows those buttons again. so not the *worst* but if it could go back to the Categorise tab, not those buttons again, that would be great...

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 03:15 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 03:51 UTC**
NOTE: extracting a thumbnail from other video services is particularly hard because oembed gives a URL - that would need to be downloaded and added somehow

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 03:57 UTC**
more notes on thumbnail:

no hotlinking?
too unreliable?

2:55
sstainsby
we could do that, sotring it in a different field i suppose
actually, just render it on the fly
that could work nicely

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 04:28 UTC**
all items but 4. (listings) have been done, anna to test on next restart of staging

sam working on listings now

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 06:32 UTC**
excellent! all those bugs squashed.

just added a couple more tasks

13. this video view page needs formatting correctly as per other videos:
http://staging.engagemedia.org:9080/Members/anna/videos/aquaporko/

14. apply plumi_workflow as per normally uploaded videos

15. thumbnails - use default image OR make required field

**Comment by mgogoulos** [comment]
**Modified by anna on 17 Dec 2014 07:03 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 07:22 UTC**
Just added a default thumbnail for EngageMedia skin "default-thumbnail".png/gif

**Comment by mgogoulos** [comment]
**Comment by anna on 17 Dec 2014 07:32 UTC**
Just added a default thumbnail for Plumi skin "plumi-default-thumbnail.jpg"

**Comment by mgogoulos** [comment]
**Comment by anna on 23 Dec 2014 00:35 UTC**
bug on external video publish
i filled out the whole form
and then clicked save changes at the bottom of Categorise tab
and it didn't save the object it took me back to here
http://staging.engagemedia.org:9080/Members/anna/videos/@@publish_video
and i can see Upload File Paste Link
ah - i see what happened
error validation
my short summary was too long
but i couldn't see my page of info, just the buttons

see screenshot

**Comment by mgogoulos** [comment]
**Modified by anna on 23 Dec 2014 00:39 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 23 Dec 2014 01:31 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 23 Dec 2014 02:37 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 23 Dec 2014 02:41 UTC**

**Comment by mgogoulos** [comment]
**Modified by anna on 23 Dec 2014 04:04 UTC**

**Comment by mgogoulos** [comment]
**Comment by sam on 23 Dec 2014 06:20 UTC**
OK, an number of issues, hopefully minor, prevented this from being deployed today:

1. /latestvideos - this view works on my local Plumi both with and without the em.skin applied; frustratingly, this is not the case on staging, as it still omits external videos. At this point we don't know why.

2. Ditto for: "Other videos by this author listing - video view pages http://staging.engagemedia.org:9080/Members/AIPP/videos/Our-Forest-Our-Life-English/view".. I strongly suspect this to be fixed when 1 is fixed.

3. The default view for an external video is fairly plain and doesn't match the appearance of the uploaded video view. It uses virtually the same template as uploaded videos, but I believe Diazo is not detecting that and applying anything to it. That should be easy to fix for someone that knows Diazo well. Alternatively, I thought about trying to use the uploaded videos 'video_view' (instead of the one I created: 'external_video_view') .. however that template needs a real uploaded file otherwise it breaks. Possibly though it could be modified to work for both - I haven't investigated how hard that would be.

4. Some validation errors don't create an 'Error:' box. Currently, the publish form needs that error box to detect that the form has been posted once and so to hide the big 'Upload' and 'Paste Link' buttons and show the form instead when the page loads. I've fixed three of these but apparently adding too much text to the summary creates an error that doesn't trigger the error box (there is just a red outline around the relevant field). Two possible resolutions: (1) fix the fields with non-standard validation behaviours, or (2) find a more robust way of letting javascript know the form has errors (eg. render something hidden) .. or both ideally.

5. Workflow - the PlumiVideo workflow isn't being applied even though PlumeExternalVideo implements the IPlumiVideo interface. In particular, the external video starts off in the 'private' state No ideas why as yet. There is no simple workflow.xml in the profile. However, considering parts of the PlumiVideo workflow involve processing, it might just be better to use a variation of Plone simple publishing workflow, modified to make the initial state 'pending'.

4. Duration - even though the video templates omit the duration box if there is none, it appears Diazo adds a box anyway and you end up with a dot which is probably a collapsed border. This is likely best fixed at the Diazo level.

**Comment by mgogoulos** [comment]
**Comment by anna on 23 Dec 2014 07:40 UTC**
note: there's more to Plumi workflow, so let's make sure we replicate that

**Comment by mgogoulos** [comment]
**Comment by sam on 15 Apr 2015 07:50 UTC**
Updates from today:

1. Fixed by just including the external video type in the relevant collection.

2. No work done on this yet.

3. This appears to be fixed by 5 on my local copy, but didn't fix the staging site. More work to come next time.

4. No work done on this yet.

5. Fixed. Checked into a new plumi.app branch called 'externalvideo'.

6. (was 4!): Duration - No work done on this yet.

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 01:11 UTC**
New problem:

7. Setting an external video to 'Featured' breaks the home page. (`AttributeError: @@embed_view`)

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 03:49 UTC**
Issue 3 is now fixed.

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 03:52 UTC**
Adding here the issue mentioned in mailing list:

8.After doing some work on the external video feature, namely changing the
workflow of external videos to be the plumi workflow, I put those
changes into action by deactivating and reactivating plumi.app ('Plumi')
in the control panel for the staging site. This causes an immediate bug
with an exception thrown from plumi.content's brain_rendering.py

Module plumi.content.browser.brain_rendering, line 98, in video_tag
IndexError: list index out of range


this comes from the line 'video = '... below :
def video_tag(self):
   path =
self.video.url[  self.request['width'](len(self.__parent__.context.absolute_url())+1:]+'/@@embed_view'
) = 525
   html = self.__parent__.context.restrictedTraverse(path)(self.request)

   video = '<video' + html.split('<video')[id="portal-footer">')[0](1].split('<div
id="portlets-footer"')[0].split('<div)
   return video


.. and the root cause seems to be that the @@embed_view view is
returning an empty string. eg.

http://staging.engagemedia.org:9080/Members/AIPP/videos/Our-Forest-Our-Life-Burmese.mp4.mp4/@@embed_view

-> (nothing)

... leading to the string manipulation code to fail.

 I suspect that view comes from collective.transcode.star, but
reinstalling that seems to have no effect.

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 05:04 UTC**
Issue 8 is fixed. It seems that @@embed_view is only populated for transcoded videos, so this is a general bug. We've fixed this by avoiding the string manipulation when the @@embed_view view returns nothing.

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 05:33 UTC**
Issue 7 is fixed .. 'featured' external videos will no longer break the home page.

**Comment by mgogoulos** [comment]
**Comment by anna on 29 Apr 2015 05:54 UTC**
2, 4 and 6 remain:

2     this view works on my local Plumi both with and without the em.skin applied; frustratingly, this is not the case on staging, as it still omits external videos. At this point we don't know why.  "Other videos by this author listing - video view pages  http://staging.engagemedia.org:9080/Members/AIPP/videos/Our-Forest-Our-Life-English/view"..

4 Some validation errors don't create an 'Error:' box. Currently, the publish form needs that error box to detect that the form has been posted once and so to hide the big 'Upload' and 'Paste Link' buttons and show the form instead when the page loads. I've fixed three of these but apparently adding too much text to the summary creates an error that doesn't trigger the error box (there is just a red outline around the relevant field). Two possible resolutions: (1) fix the fields with non-standard validation behaviours, or (2) find a more robust way of letting javascript know the form has errors (eg. render something hidden) .. or both ideally. 

this is inconvenient/confusing in some circumstances but probably not critical

6. Duration - even though the video templates omit the duration box if there is none, it appears Diazo adds a box anyway and you end up with a dot which is probably a collapsed border. This is likely best fixed at the Diazo level. 

this is cosmetic and likely very easy 

Sam is now proceeding with number 2.

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 06:21 UTC**
2 is fixed.

**Comment by mgogoulos** [comment]
**Modified by anna on 29 Apr 2015 06:49 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 29 Apr 2015 06:52 UTC**

18. issue with other videos by this author - no image or duration
http://staging.engagemedia.org:9080/Members/anna/videos/another-text/view

see attached screenshot.

**Comment by mgogoulos** [comment]
**Modified by anna on 29 Apr 2015 06:58 UTC**

**Comment by mgogoulos** [comment]
**Comment by anna on 29 Apr 2015 07:03 UTC**
OK, seems we have had a little confusion with numbers in this ticket, but here is a consolidation of remaining tasks from the description with those remaining from Sam's comments, and also from another bug I just found.

So currently looking at six live issues.

== REMAINING TASKS ==
 

1. Ensure videos appear in listings throughout the site as consistent with uploaded videos - titles, thumbnails, duration etc.

CONFIRMED WORKING

 * Front page latest videos
 * http://staging.engagemedia.org/
 * Members video folder
 * http://staging.engagemedia.org:9080/Members/anna/videos
 * Author page
 * http://staging.engagemedia.org:9080/author/anna
 * Latest videos page
 * http://staging.engagemedia.org:9080/latestvideos
 * Other videos by this author listing - video view pages
 * http://staging.engagemedia.org:9080/Members/AIPP/videos/Our-Forest-Our-Life-English/view
 * Search results

NOT WORKING YET (TESTED):

 * Taxonomy listings - Genre, Topic, Country, Language - THESE EXIST IN THE EM.TAXONOMIES AND EM.SKIN PRODUCTS
 * eg. http://staging.engagemedia.org:9080/taxonomy/genre/documentary

4. error validation - no message on short summary, brings you back to the buttons not to the page itself . Some validation errors don't create an 'Error:' box. Currently, the publish form needs that error box to detect that the form has been posted once and so to hide the big 'Upload' and 'Paste Link' buttons and show the form instead when the page loads. I've fixed three of these but apparently adding too much text to the summary creates an error that doesn't trigger the error box (there is just a red outline around the relevant field). Two possible resolutions: (1) fix the fields with non-standard validation behaviours, or (2) find a more robust way of letting javascript know the form has errors (eg. render something hidden) .. or both ideally.

this is inconvenient/confusing in some circumstances but probably not critical

6. Duration - even though the video templates omit the duration box if there is none, it appears Diazo adds a box anyway and you end up with a dot which is probably a collapsed border. This is likely best fixed at the Diazo level. 

this is cosmetic and likely very easy 

14. apply '''plumi_workflow''' as per normally uploaded videos - put it in the profile - check and copy across anything else from profile including the views

17. move thumbnail attachment to underneath add video file - and add the red dot for required field (even if it is not *actually* required). Change helper text ONLY FOR EXTERNAL VIDEOS to say "Add a thumbnail image for this video".

18. issue with other videos by this author - no image or duration 

 http://staging.engagemedia.org:9080/Members/anna/videos/another-text/view

see attached screenshot.

**Comment by mgogoulos** [comment]
**Comment by anna on 29 Apr 2015 07:16 UTC**
note: was getting 502 errors when loading pages towards end of our day

and strange response when loading front page:
"Not enough data in request or socket error"

check in on this next week...

**Comment by mgogoulos** [comment]
**Comment by sam on 29 Apr 2015 07:17 UTC**
Quick note: Issue 6 is caused by a div with class"duration" being added by `em/skin/theme_resources/rules.xml` even if there isn't a duration.

**Comment by mgogoulos** [comment]
**Comment by sam on 6 May 2015 00:23 UTC**
I believe 14 is already done - probably needs a fair bit of testing.

**Comment by mgogoulos** [comment]
**Comment by sam on 6 May 2015 01:07 UTC**
Item 6 (duration display) is done.

**Comment by mgogoulos** [comment]
**Comment by sam on 6 May 2015 05:53 UTC**
Item 1 is completed as taxonomy listing are now fixed.

**Comment by mgogoulos** [comment]
**Comment by anna on 6 May 2015 06:20 UTC**
so only 4, 17, 18 remain - description has been updated

**Comment by mgogoulos** [comment]
**Comment by sam on 20 May 2015 02:10 UTC**
Point 17 is implemented and ready for testing.

**Comment by mgogoulos** [comment]
**Comment by anna on 20 May 2015 02:51 UTC**
tried to test this but testing site is returning 502 on publishing external videos. no space for blob storage.

**Comment by mgogoulos** [comment]
**Comment by anna on 20 May 2015 04:01 UTC**
Regarding testing of 17:

The form looks good with re-positioned thumbnail and required dot. 

However - issue 4 is provoked when uploading *without* a thumbnail.

And a new issue that must have cropped up during recent changes

19. When uploading a video of 681 kB, the file appeared to successfully upload using widget BUT an error was thrown

Error
    There were some errors.

This error is obviously unspecified. Difficult to test properly with jammed blob storage - so perhaps is due to this... 

in the log: Products.MailHost.MailHost.MailHostError: No message recipients designated <-- I guess disabling the email notifications also causes errors :-(

See ticket here about notifications errors:
http://trac.plumi.org/ticket/992#comment:1

**Comment by mgogoulos** [comment]
**Comment by sam on 20 May 2015 04:19 UTC**
Point 4 is implemented and ready for testing.

**Comment by mgogoulos** [comment]
**Comment by sam on 20 May 2015 04:21 UTC**
PS: the fix for 4 should pretty much cover any cases with verification errors that don't create an 'Error:' box at the top of the form .. not just for the summary text. Ideally, all such errors should create an error box like that, but that's a different issue not related to  external videos.

**Comment by mgogoulos** [comment]
**Modified by anna on 20 May 2015 04:56 UTC**

**Comment by mgogoulos** [comment]
**Comment by sam on 20 May 2015 06:45 UTC**
Trying to use `hasThumbnailImage` in `VideoView` to fix 18. However, it appears to always return `True` for any kind of video, so this will need to be fixed.

**Comment by mgogoulos** [comment]
**Comment by anna on 20 May 2015 08:38 UTC**
Tried to test a few times but getting 504 errors on publishing videos. Will wait until new test site is set up.

**Comment by mgogoulos** [comment]
**Comment by anna on 10 Nov 2016 07:25 UTC**
This is very close, however as work has stalled for now, we should also explore this option:

https://pypi.python.org/pypi/sc.embedder

I suppose the problem is - how to include it within listings and taxonomy

**Comment by mgogoulos** [comment]
**Comment by anna on 29 Nov 2016 05:21 UTC**
Pushing this to a subsequent release.
